src/main/java/br/com/deveficiente/nossomercadolivreapi/categoria/VerificaSeCategoriaSuperiorExisteValidator.java

```diff
@@ -24,7 +25,8 @@ public void validate(Object o, Errors errors) {
         if (categoriaRequest.getCategoriaSuperiorId() == null || categoriaRequest.getCategoriaSuperiorId() <= 0) {
```

Comment:
*Usamos tudo que conhecemos que está pronto. Só fazemos código do zero se for estritamente necessário.*

A primeira validação que roda é a da bean validation, é um detalhe de implementação que não imagino o Spring mudando tão cedo. Por conta disso, você pode simplesmente checar com ``èrrors.hasErrors()``` antes de continuar a execução da validador.

Reply:
Oi Alberto, não entendi essa parte.

O campo categoriaSuperiorId não é obrigatório, não tem nenhuma anotação sobre ele, por isso pode vir nulo, então a invocação do erros.hasErrors() retornaria false neste caso.

Reply:
Putz, é verdade, foi mal. Então, dado que esse código é necessário, vamos a seguinte regra :P

*Não retornamos nulo dentro das regras da aplicação. Pense que seu computador vai explodir.*

Então aqui você pode fazer o método retornar Optional, ou criar um método no objeto que faz a checagem para você :).